debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/rest/DebeziumMongoDbConnectorResource.java

```diff
@@ -51,24 +46,9 @@ public Connector getConnector() {
         return new MongoDbConnector();
     }
 
-    protected MongoDbConnection getConnection(Configuration configuration) {
-        MongoDbTaskContext context = new MongoDbTaskContext(configuration);
-        ReplicaSet replicaSet = new ReplicaSet(context.getConnectionContext().connectionString());
-        return context.getConnectionContext().connect(replicaSet, context.filters(), (s, throwable) -> {
-            throw new DebeziumException(s, throwable);
-        });
-    }
-
     @Override
     public List<DataCollection> getMatchingCollections(Configuration configuration) {
-        try (MongoDbConnection connection = getConnection(configuration)) {
-            return connection.collections().stream()
-                    .map(collectionId -> new DataCollection(collectionId.replicaSetName(), collectionId.dbName(), collectionId.name()))
-                    .collect(Collectors.toList());
-        }
-        catch (InterruptedException e) {
-            throw new DebeziumException(e);
-        }
+        return ((MongoDbConnector) getConnector()).getMatchingCollections(configuration);
```

Comment:
@jpechane Would it maybe make sense to add another interface on the connector classes/implementations and let `getConnector()` return this one here? That way we could move this code here into the FilterValidationResource class instead of having this duplicate coder everywhere and an unnecessary cast. Wdyt?

Reply:
Is the typecasting even necessary? IMHO it could be pulled-up to the parent class and just call `return getConnector().getMatchingCollections(configuration);`

Reply:
ok, I see what you mean. We don't have a common parent for all Debezium connectors. Maybe instead of adding a new interface we should add a common parent class for all connectors lakie we do for tasks so intorducing `BaseConnector` class?

Reply:
Agreed!